flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -1488,6 +1489,20 @@
     // JSON functions
     // --------------------------------------------------------------------------------------------
 
+    public static final BuiltInFunctionDefinition IS_JSON =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("IS_JSON")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            or(
+                                    sequence(logical(LogicalTypeFamily.CHARACTER_STRING)),
+                                    sequence(
+                                            logical(LogicalTypeFamily.CHARACTER_STRING),
+                                            symbol(JsonType.class))))
+                    .outputTypeStrategy(explicit(DataTypes.BOOLEAN().notNull()))
```

Comment:
shouldn't this be `nullifArgs`? Maybe the docs are not clear enough for me as mentioned above.

Reply:
`IS JSON` is never nullable, it returns boolean even on null input.

Reply:
`IS JSON` is supposed to never be nullable, and always return true or false. However, it does seem this is currently implemented incorrectly. :-/

Reply:
I'll fix that as part of this PR.